### ISSUE TITLE
Upgrade to sklearn 1.0+

### ISSUE THREAD

**Issue by GillesVandewiele** [tslearn-team/tslearn#375]
sklearn has recently published a new major version: 1.0.0, we should upgrade our dependency as well (and check for backward compatibility) and make sure all of our tests comply to the new API changes.

**Comment by rtavenar** [comment]
Could you summarize the main API changes here ?
I think that all estimators should have default values for all their hyper parameters right ?

**Comment by GillesVandewiele** [comment]
I haven't really looked into the release notes into detail, but I will definitely make a exhaustive list of changes here first before I start adapting the code. I will probably only have time next weekend (after 10 December), but could shift in my planning if this is rather urgent?

**Comment by rtavenar** [comment]
I do not thing there is any emergency here (I mean tslearn code still works OK, we just don't match the latest specifications), Dec 10 should be great!

**Comment by rtavenar** [comment]
I think @GillesVandewiele is the one who knows...

**Comment by rtavenar** [comment]
Now merged to `main` branch!
